pkg/agent/cniserver/pod_configuration.go

```diff
@@ -340,37 +366,55 @@ func removeInterfaces(
 	ovsPortName := containerConfig.IfaceName
 	klog.V(2).Infof("Deleting OVS port with UUID %s peer container %s", portUUID, containerID)
 	// Remove openflow entries of target container
-	if err := ofClient.UninstallPodFlows(ovsPortName); err != nil {
+	if err := pc.ofClient.UninstallPodFlows(ovsPortName); err != nil {
 		klog.Errorf("Failed to delete Openflow entries for container %s: %v", containerID, err)
 		return err
 	}
 	// TODO: handle error and introduce garbage collection for failure on deletion
-	if err := ovsBridgeClient.DeletePort(portUUID); err != nil {
+	if err := pc.ovsBridgeClient.DeletePort(portUUID); err != nil {
 		klog.Errorf("Failed to delete OVS port %s: %v", portUUID, err)
 		return err
 	}
 	// Remove container configuration from cache.
-	ifaceStore.DeleteInterface(ovsPortName)
+	pc.ifaceStore.DeleteInterface(ovsPortName)
 	klog.Infof("Interfaces removed successfully for container %s", containerID)
 	return nil
 }
 
-func checkInterfaces(ifaceStore agent.InterfaceStore, containerID string, containerNetNS string, containerIface, hostIface *current.Interface, hostVethName string, prevResult *current.Result) error {
+func (pc *podConfigurator) checkInterfaces(
+	containerID, containerNetNS, hostVethName string,
+	containerIface, hostIface *current.Interface,
+	prevResult *current.Result) error {
 	netns, err := ns.GetNS(containerNetNS)
 	if err != nil {
 		klog.Errorf("Failed to check netns config %s: %v", containerNetNS, err)
 		return err
 	}
 	defer netns.Close()
-	if containerlink, err := checkContainerInterface(netns, containerNetNS, containerID, containerIface, prevResult); err != nil {
+	if containerlink, err := checkContainerInterface(
+		containerNetNS,
+		containerID,
+		netns,
+		containerIface,
+		prevResult); err != nil {
 		return err
-	} else if err := checkHostInterface(ifaceStore, hostIface, containerIface, containerlink, containerID, hostVethName, prevResult.IPs); err != nil {
+	} else if err := pc.checkHostInterface(
+		containerID,
+		hostVethName,
+		hostIface,
+		containerIface,
+		containerlink,
+		prevResult.IPs); err != nil {
 		return err
 	}
 	return nil
 }
 
-func checkContainerInterface(netns ns.NetNS, containerNetns, containerID string, containerIface *current.Interface, prevResult *current.Result) (*vethPair, error) {
+func checkContainerInterface(
```

Comment:
should this be method of podConfigurator too, like checkHostInterface?

Reply:
For now, I only changed funcs that access podConfigurator fields to methods. Would you expect to change all funcs in this file to methods?

Reply:
Not all funcs but those have similar effects, for example it looks strange to me that: 
`checkHostInterface` has a receiver but `checkContainerInterface` does not;
`removeInterfaces` and `checkInterfaces` have a receiver but `setupInterface` does not.

Those util like functions `buildContainerConfig` and `parseContainerIP` could be kept as is IMO.

Reply:
Ok. Changed this two.